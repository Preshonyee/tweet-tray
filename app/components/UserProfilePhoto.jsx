import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';
import Theme from 'styled-theming';

import * as constants from '../constants';

import ProgressArc from './ProgressArc';

const UserProfilePhotoStyle = Styled.div`
    overflow: hidden;
    width: 58px;
    height: 58px;
    position: absolute;
    left: 12px;
    top: 17px;
`;

const ProfilePhotoImageStyle = Styled.div`
    position: relative;
    top: 3px;
    left: 3px;
    border-radius: 50px;
    overflow: hidden;
    width: 52px;
    height: 52px;
    background-color: ${Theme('mode', { day: constants.LIGHT_GREY, night: constants.DARK_MODE_FOREGROUND, })};
`;

const WordCounterStyle = Styled.div`
    position: absolute;
    height: 60px;
    width: 60px;
    top: -1px;
    left: -1px;

    .SVG {
        height: 100%;
        width: 100%;
        margin: 0px;
        position: relative;

        circle {
          stroke-width: 8px !important;
        }
      }
    }
`;

class UserProfilePhoto extends Component {
  static propTypes = {
    profilePhotoURL: PropTypes.string,
    weightedTextAmount: PropTypes.number,
  }

  static defaultProps = {
    profilePhotoURL: null,
    weightedTextAmount: null,
  }

  constructor(props) {
    super(props);

    this._calculateArcValue = this._calculateArcValue.bind(this);
    this._calculateArcColor = this._calculateArcColor.bind(this);
  }

  _calculateArcColor() {
    const { weightedTextAmount, } = this.props;

    let color = null;

    if (weightedTextAmount !== null) {
      if (weightedTextAmount > 1000) {
        color = constants.RED;
      } else {
        color = constants.BLUE;

        if (weightedTextAmount >= 900) {
          color = constants.YELLOW;
        }
      }
    } else {
      color = constants.BLUE;
    }

    return color;
  }

  _calculateArcValue() {
    const { weightedTextAmount, } = this.props;

    let arcValue = 0;

    if (weightedTextAmount !== null) {
      if (weightedTextAmount > 1000) {
        arcValue = parseInt(100, 10);
      } else {
        arcValue = parseInt((weightedTextAmount / 10), 10);
      }
    }

    return arcValue;
  }

  render() {
    const { profilePhotoURL, } = this.props;

    return (
      <UserProfilePhotoStyle>
        <ProfilePhotoImageStyle>
          <img src={profilePhotoURL} alt="Profile" width="52" height="52" />
        </ProfilePhotoImageStyle>
        <WordCounterStyle>
          <ProgressArc
            textVisible={false}
            arcColor={this._calculateArcColor()}
            rounded
            arcBackgroundColor="transparent"
            className="SVG"
            value={this._calculateArcValue()}
          />
        </WordCounterStyle>
      </UserProfilePhotoStyle>
    );
  }
}

export default UserProfilePhoto;
