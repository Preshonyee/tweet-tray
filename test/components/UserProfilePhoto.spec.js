import React from 'react';
import Enzyme, { shallow, } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserProfilePhoto from '../../app/components/UserProfilePhoto';
import * as testData from '../testData';

Enzyme.configure({adapter: new Adapter()});

function setup() {
  const component = shallow(<UserProfilePhoto
    profilePhotoURL={testData.userCredentials.profileImageURL}
    weightedTextAmount={testData.weightedStatus.permillage}
  />);
  return {
    component,
    image: component.find('img'),
    arc: component.find('.SVG'),
  };
}

describe('UserProfilePhoto component', () => {
  it('should display profile image', () => {
    const { image, } = setup();
    expect(image.find("img").prop("src")).toMatchSnapshot();
  });

  it('should display word counter arc with 74% filled', () => {
    const { arc, } = setup();
    expect(arc.prop('value')).toMatchSnapshot();
  });
});
