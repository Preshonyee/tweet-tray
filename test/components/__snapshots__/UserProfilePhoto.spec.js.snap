// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfilePhoto component should display profile image 1`] = `"https://pbs.twimg.com/profile_images/960374717726130182/AS3TkK0J_normal.jpg"`;

exports[`UserProfilePhoto component should display word counter arc with 74% filled 1`] = `74`;
